tests/ttnn/unit_tests/tensor/test_tensor_creation.py

```diff
@@ -120,3 +120,143 @@ def test_tensor_creation_api_parity(shape, tt_dtype, layout, device):
     passing = torch.allclose(py_tensor, py_tensor_after_round_trip_3, **allclose_kwargs)
     passing = torch.allclose(py_tensor, py_tensor_after_round_trip_4, **allclose_kwargs)
     assert passing
+
+
+grid_size = [8, 7]
+
+
+@pytest.mark.parametrize(
+    "layout, tile",
+    [
+        (ttnn.ROW_MAJOR_LAYOUT, None),
+        (ttnn.TILE_LAYOUT, ttnn.Tile([32, 32])),
+        (ttnn.TILE_LAYOUT, ttnn.Tile([16, 16])),
+        (ttnn.TILE_LAYOUT, ttnn.Tile([32, 16])),
+        (ttnn.TILE_LAYOUT, ttnn.Tile([16, 16])),
+    ],
+)
+@pytest.mark.parametrize(
+    "tt_dtype",
+    [
+        ttnn.uint8,
+        ttnn.uint16,
+        ttnn.uint32,
+        ttnn.int32,
+        ttnn.float32,
+        ttnn.bfloat16,
+        ttnn.bfloat8_b,
+        ttnn.bfloat4_b,
+    ],
+)
+@pytest.mark.parametrize(
+    "shape, memory_config",
+    [
+        (
+            (1, 2, 3, 4),
```

Comment:
can we add tests for smaller ranks too?

Reply:
Intent was to make sure I could represent interleaved in the same path (with potential padding if needed) as well. I could add another test with more concise parameters that sweeps ranks as well.